### ISSUE TITLE
Current Python version 3.11.8，Error

### ISSUE THREAD

**Issue by LPG-VERSION** [KohakuBlueleaf/PixelOE#2]
.\python_embeded\python.exe -s -m pip install pixeloe
Collecting pixeloe
  Using cached pixeloe-0.0.7.tar.gz (13 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\Administrator\AppData\Local\Temp\pip-install-4wi6wlka\pixeloe_735c229b01b14ca58913a932eaa95fbb\setup.py", line 2, in <module>
          from pixeloe.cli import command_map
      ModuleNotFoundError: No module named 'pixeloe'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

**Comment by KohakuBlueleaf** [comment]
OH I'm stupid LoL
Will fix this

**Comment by KohakuBlueleaf** [comment]
You can try to install it from source
pypi installation will be fixed in near future

**Comment by CreateWorld-Animation** [comment]
ok， let me have a try

**Comment by KohakuBlueleaf** [comment]
@LPG-VERSION This should be fixed now (0.0.8)

**Comment by YanWenKun** [comment]
```
  Downloading pixeloe-0.0.9.tar.gz (13 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  python setup.py egg_info did not run successfully.
  exit code: 1
  
  [6 lines of output]
  Traceback (most recent call last):
    File "<string>", line 2, in <module>
    File "<pip-setuptools-caller>", line 34, in <module>
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-install-86urpfsf\pixeloe_5512c8b144a24128b11d7851d7fe7a2b\setup.py", line 2, in <module>
      from pixeloe import cli
  ModuleNotFoundError: No module named 'pixeloe'
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
```

I guess it could be:

```diff
setup.py

- from pixeloe import cli
+ from .pixeloe import cli
```

**Comment by KohakuBlueleaf** [comment]
> ```
>   Downloading pixeloe-0.0.9.tar.gz (13 kB)
>   Preparing metadata (setup.py): started
>   Preparing metadata (setup.py): finished with status 'error'
>   error: subprocess-exited-with-error
>   
>   python setup.py egg_info did not run successfully.
>   exit code: 1
>   
>   [6 lines of output]
>   Traceback (most recent call last):
>     File "<string>", line 2, in <module>
>     File "<pip-setuptools-caller>", line 34, in <module>
>     File "C:\Users\runneradmin\AppData\Local\Temp\pip-install-86urpfsf\pixeloe_5512c8b144a24128b11d7851d7fe7a2b\setup.py", line 2, in <module>
>       from pixeloe import cli
>   ModuleNotFoundError: No module named 'pixeloe'
>   [end of output]
>   
>   note: This error originates from a subprocess, and is likely not a problem with pip.
> error: metadata-generation-failed
> ```
> 
> I guess it could be:
> 
> ```diff
> setup.py
> 
> - from pixeloe import cli
> + from .pixeloe import cli
> ```

Umm I don't think so 
https://stackoverflow.com/questions/45280653/import-a-script-from-package-in-setup-py

It looks like we should not put any relative import

Another thing is the path of setup.py is not a package, so it is definitely not "import .pixeloe"

**Comment by LPG-VERSION** [comment]
`
![Uploading image.png…]()
`

**Comment by CreateWorld-Animation** [comment]
from setuptools import setup, find_packages
from pixeloe import command_map

setup(
    name="pixeloe",
    packages=find_packages(),
    version="0.0.7",
    url="https://github.com/KohakuBlueleaf/PixelOE",
    description="Detail-Oriented Pixelization based on Contrast-Aware Outline Expansion.",
    license="Apache License 2.0",
    author="Shih-Ying Yeh(KohakuBlueLeaf)",
    author_email="apolloyeh0123@gmail.com",
    zip_safe=False,
    install_requires=["opencv-python", "numpy", "pillow"],
    entry_points={
        "console_scripts": [
            f"pixeloe.{command}=pixeloe.cli:{func}"
            for command, func in command_map.items()
        ]
    },
    python_requires=">=3.10",
)

**Comment by LPG-VERSION** [comment]
应该这样改,还有根目录缺少__init__.py

**Comment by CreateWorld-Animation** [comment]
It should be changed like this, and the root directory is missing __ init__. py

**Comment by CreateWorld-Animation** [comment]
Relative import can directly import sub modules from the parent module

**Comment by KohakuBlueleaf** [comment]
> It should be changed like this, and the root directory is missing __ init__. py

Then that's not what I need here.

**Comment by KohakuBlueleaf** [comment]
Root dir should not have __init__.py

**Comment by KohakuBlueleaf** [comment]
I will use other method to fix this, but definitely not put `__init__.py` in root dir

**Comment by CreateWorld-Animation** [comment]
But the entire project is a dependency package, and the root directory does not have __ init__. py, so the pixloe package will not be recognized by the environment. I have tested it before

**Comment by CreateWorld-Animation** [comment]
Okay, I'll wait for you to make the changes. Come on, adventurer

**Comment by KohakuBlueleaf** [comment]
> But the entire project is a dependency package, and the root directory does not have __ init__. py, so the pixloe package will not be recognized by the environment. I have tested it before

Your understanding is incorrect.
Only the pixeloe folder is the package we need.
My implementation is wrong, but make whole project a package is worst option.

**Comment by YanWenKun** [comment]
> Umm I don't think so https://stackoverflow.com/questions/45280653/import-a-script-from-package-in-setup-py
> 
> It looks like we should not put any relative import
> 
> Another thing is the path of setup.py is not a package, so it is definitely not "import .pixeloe"

啊！不好意思之前回复得太草率了（遇到过类似情况，但这次不一样）

我大概明白问题原因了，这个包既提供 API (作为其他包依赖的 lib），同时也是 CLI 应用，CLI 的部分在安装时遇到了“找不到我自己”的问题。
这在我熟悉的领域一般是分开的，比如 `curl` 和 `libcurl`。不过似乎在 Python 领域没有这个必要，比如 `huggingface_hub` 就兼顾了两者。

我是在打 [Windows 整合包](https://github.com/YanWenKun/ComfyUI-Windows-Portable) 时报的错，[构建 Docker 镜像](https://github.com/YanWenKun/ComfyUI-Docker) 则顺利无问题。

不太懂 Python，暂时帮不上什么忙，只能先报个错，有劳青叶大大了！

---

(English translation by ChatGPT)

Ah! Sorry for the hasty response earlier (I've encountered similar situations before, but this one is different).

I think I understand the reason for the problem now. This package provides both an API (as a lib that other packages depend on) and also a CLI application. The issue arises during installation when it can't find itself.

In the field I'm familiar with, these are usually separated, like `curl` and `libcurl`. However, it seems in the Python domain, this separation might not be necessary, as seen with `huggingface_hub`, which combines both.

I encountered this error while building the [Windows standalone package](https://github.com/YanWenKun/ComfyUI-Windows-Portable), but had no issues with [building the Docker image](https://github.com/YanWenKun/ComfyUI-Docker).

I don't know much about Python, so I can't help much for now. Just reporting the error. Thanks a lot in advance!

**Comment by YanWenKun** [comment]
Well... I guess this is not elegant enough but it works for my installation on Windows:

```diff
--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,4 @@
 from setuptools import setup, find_packages
-from pixeloe import cli
 
 
 setup(
@@ -15,8 +14,8 @@ setup(
     install_requires=["opencv-python", "numpy", "pillow"],
     entry_points={
         "console_scripts": [
-            f"pixeloe.{command}=pixeloe.cli:{func}"
-            for command, func in cli.command_map.items()
+            "pixeloe.pixelize=pixeloe.cli:pixelize",
+            "pixeloe.outline=pixeloe.cli:outline"
         ]
     },
     python_requires=">=3.10",

```

I kinda understand why this works. But I really don't know why `pip` or `setuptools` on Linux won't throw `No module named 'pixeloe'` error. Looks like magic to me, at least for today.

**Comment by bigcat88** [comment]
have the same problem on windows with python 3.10

```
from pixeloe import cli
  ModuleNotFoundError: No module named 'pixeloe'
  ```

**Comment by Kidel** [comment]
> Well... I guess this is not elegant enough but it works for my installation on Windows:
> 
> ```diff
> --- a/setup.py
> +++ b/setup.py
> @@ -1,5 +1,4 @@
>  from setuptools import setup, find_packages
> -from pixeloe import cli
>  
>  
>  setup(
> @@ -15,8 +14,8 @@ setup(
>      install_requires=["opencv-python", "numpy", "pillow"],
>      entry_points={
>          "console_scripts": [
> -            f"pixeloe.{command}=pixeloe.cli:{func}"
> -            for command, func in cli.command_map.items()
> +            "pixeloe.pixelize=pixeloe.cli:pixelize",
> +            "pixeloe.outline=pixeloe.cli:outline"
>          ]
>      },
>      python_requires=">=3.10",
> ```
> 
> I kinda understand why this works. But I really don't know why `pip` or `setuptools` on Linux won't throw `No module named 'pixeloe'` error. Looks like magic to me, at least for today.

This worked for me.

**Comment by belladoreai** [comment]
I have the same problem. I'm on Windows, python 3.11.6.

**Comment by aiisoft** [comment]
Collecting pixeloe
  Using cached pixeloe-0.0.9.tar.gz (13 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\HP\AppData\Local\Temp\pip-install-ry8zttp0\pixeloe_feadb1b401564c5cb9613c0d65a23b12\setup.py", line 2, in <module>
          from pixeloe import cli
      ModuleNotFoundError: No module named 'pixeloe'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

**Comment by KohakuBlueleaf** [comment]
Fixed in 0.0.10
